api/models/Notification.js

```diff
@@ -89,11 +90,25 @@ module.exports = bookshelf.Model.extend({
         return this.sendJoinRequestPush()
       case 'approvedJoinRequest':
         return this.sendApprovedJoinRequestPush()
+      case 'announcement':
+        return this.sendPushAnnouncement()
       default:
         return Promise.resolve()
     }
   },
 
+  sendPushAnnouncement: function (version) {
+    var post = this.post()
+    var communityIds = Activity.communityIds(this.relations.activity)
+    if (isEmpty(communityIds)) throw new Error('no community ids in activity')
+    return Community.find(communityIds[0])
+    .then(community => {
+      var path = url.parse(Frontend.Route.post(post, community)).path
+      var alertText = PushNotification.textForPost(post, community, this.relations.activity.get('reader_id'), version)
```

Comment:
Do you want the announcement notification text to be different in any way from a normal post? Probably. Even if it's just appending "Announcement: " to the front. If so you will need to either expand textForPost or create a different method (PushNotification.textForAnnouncement) to be used here.